src/libraries/System.Private.CoreLib/src/System/Guid.cs

```diff
@@ -1421,14 +1421,23 @@ private static (Vector128<byte>, Vector128<byte>, Vector128<byte>) FormatGuidVec
                     Vector128.Create(0x7060504FF030201, 0xF0E0D0C0B0A0908).AsByte());
 
                 // Vector "z" - we need to merge some elements of hexLow with hexHigh and add 4 dashes.
-                Vector128<byte> mid1 = Vector128.Shuffle(hexLow,
-                    Vector128.Create(0x0D0CFF0B0A0908FF, 0xFFFFFFFFFFFF0F0E).AsByte());
-                Vector128<byte> mid2 = Vector128.Shuffle(hexHigh,
-                    Vector128.Create(0xFFFFFFFFFFFFFFFF, 0xFF03020100FFFFFF).AsByte());
-                Vector128<byte> dashesMask = Vector128.Shuffle(Vector128.CreateScalarUnsafe((byte)'-'),
-                    Vector128.Create(0xFFFF00FFFFFFFF00, 0x00FFFFFFFF00FFFF).AsByte());
-
-                Vector128<byte> vecZ = (mid1 | mid2 | dashesMask);
+                Vector128<byte> vecZ;
+                Vector128<byte> dashesMask = Vector128.Create(0x00002D000000002D, 0x2D000000002D0000).AsByte();
+                if (AdvSimd.Arm64.IsSupported)
```

Comment:
Can you leave a comment explaining why this path exists, probably, with a TODO about merging the lookups in JIT.

Reply:
Also I think an explanation about various hexs used in the code will be good to have for future maintenance.

Reply:
Done.